pywps/inout/basic.py

```diff
@@ -1033,3 +1036,22 @@ def get_url(self):
         (_, _, url) = self.storage.store(self)
         # url = self.storage.url(self)
         return url
+
+
+def posix_path(path: Union[str, Path]) -> str:
+    path = str(path)
+    return path.replace('\\', '/') if os.name == 'nt' else path
+
+
+def file_uri(path: Union[str, Path]) -> str:
+    path = Path(path)
+    path = path.as_uri()
+    return str(path)
+
+
+def uri_to_path(uri) -> str:
+    p = urlparse(uri)
+    path = p.path
+    if os.name == 'nt':
```

Comment:
> This might not be enough to determine if the OS/Platform is actually Windows. I would leverage `platform.system()` for a few windows-like keys. See: https://stackoverflow.com/a/54837707/7322852

I looked at the post, I don't mind to follow your recommendation, but I couldn't find any system (windows-like or not) that have `os.name != 'nt' and platform.system() == 'windows'`

Reply:
I bring it up here since depending on the subsystem that the script is running in (msys2, cygwin, WSL2, etc.), this check might not return what we intend it to. There's also a possibility that the `pathlib.Path` function handles the file path changes in a better way (see: https://docs.python.org/3/library/pathlib.html#pathlib.PureWindowsPath) e.g.:
```
from pathlib import Path, PureWindowsPath, PurePosixPath
p = Path("D/path/to/a/file")
p.as_posix()
Out[5]: 'D/path/to/a/file'
str(PureWindowsPath(p))
Out[9]: 'D\\path\\to\\a\\file'
PurePosixPath(PureWindowsPath(p))
Out[15]: PurePosixPath('D/path/to/a/file')
```

To be fair, I haven't done much in Windows. Just presenting some options.

Reply:
Thanks for bringing this up.
Actually I've used it here, but the only thing I couldn't figure out is how to get a proper WindowsPath from a file uri:
```
> p = r'c:\abc.txt'
> wp = WindowsPath(p)
> wp
WindowsPath('c:/abc.txt')
> wp.as_uri()
'file:///c:/abc.txt'
> from urllib.parse import urlparse
> parse = urlparse(uri)
> parse.path
'/c:/abc.txt'
> Path(parse.path)
WindowsPath('/c:/abc.txt')
```
How can I get rid of this leading slash without checking actually trimming it if on windows ?